src/transports/encodePolyfill.ts

```diff
@@ -1,14 +1,20 @@
-import { CAP_GLOBAL_OBJ } from '../utils/webViewUrl';
+import { getMainCarrier, SDK_VERSION } from '@sentry/core';
+import type { CAP_GLOBAL_OBJ } from 'src/utils/webViewUrl';
+
 import { utf8ToBytes } from '../vendor';
 
 export const useEncodePolyfill = (): void => {
-  if (!CAP_GLOBAL_OBJ.__SENTRY__) {
-    (CAP_GLOBAL_OBJ.__SENTRY__ as Partial<
+  const globalCarriers = getMainCarrier().__SENTRY__;
+  if (!globalCarriers) {
+    (globalCarriers as Partial<
       (typeof CAP_GLOBAL_OBJ)['__SENTRY__']
     >) = {};
   }
+  const capacitorSiblingCarrier = globalCarriers?.[SDK_VERSION];
 
-  CAP_GLOBAL_OBJ.__SENTRY__.encodePolyfill = encodePolyfill;
+  if (capacitorSiblingCarrier) {
+    capacitorSiblingCarrier.encodePolyfill = encodePolyfill;
+  }
```

Comment:
We can simplify this by copying what JS is doing.

````suggestion
  // Based on https://github.com/getsentry/sentry-javascript/blob/f0fc41f6166857cd97a695f5cc9a18caf6a0bf43/packages/core/src/carrier.ts#L49
  const carrier = getMainCarrier();
  const __SENTRY__ = (carrier.__SENTRY__ = carrier.__SENTRY__ || {});
  (__SENTRY__[SDK_VERSION] = __SENTRY__[SDK_VERSION] || {}).encodePolyfill = encodePolyfill;
````